packages/react/src/components/form/form.tsx

```diff
@@ -14,42 +14,127 @@ import {
   ConfigurationOverrides,
   ConfigurationOverridesProps,
 } from "../configuration-overrides";
-import { LoginOptions } from "../../domain/types";
+import { Handle, HandleType, LoginOptions } from "../../domain/types";
+import React, { useCallback } from "react";
+import { useSlots } from "../slot";
+import { Factor } from "@slashid/slashid";
+import { Flag } from "../input";
 
 export type Props = ConfigurationOverridesProps & {
   className?: string;
   onSuccess?: CreateFlowOptions["onSuccess"];
   middleware?: LoginOptions["middleware"];
+  children?: React.ReactNode;
 };
 
-export const Form: React.FC<Props> = ({
+type PayloadOptions = {
+  handleType?: HandleType;
+  handleValue?: string;
+  flag?: Flag;
+};
+
+export type InternalFormContextType = {
+  flowState: ReturnType<typeof useFlowState> | null;
+  lastHandle?: Handle;
+  submitPayloadRef: React.MutableRefObject<PayloadOptions>;
+  handleSubmit: (factor: Factor, handle?: Handle) => void;
+  selectedFactor?: Factor;
+  setSelectedFactor: React.Dispatch<React.SetStateAction<Factor | undefined>>;
+};
+export const InternalFormContext = React.createContext<InternalFormContextType>(
+  {
+    flowState: null,
+    lastHandle: undefined,
+    submitPayloadRef: { current: {} },
+    handleSubmit: () => null,
+    selectedFactor: undefined,
+    setSelectedFactor: () => null,
+  }
+);
+export const useInternalFormContext = () => {
+  return React.useContext(InternalFormContext);
+};
+
+export const Form = ({
   className,
   onSuccess,
   factors,
   text,
-  middleware
-}) => {
+  middleware,
+  children,
+}: Props) => {
   const flowState = useFlowState({ onSuccess });
   const { showBanner } = useConfiguration();
   const { lastHandle } = useLastHandle();
+  const submitPayloadRef = React.useRef<PayloadOptions>({
+    handleType: undefined,
+    handleValue: undefined,
+    flag: undefined,
+  });
+  const [selectedFactor, setSelectedFactor] = React.useState<
+    Factor | undefined
+  >();
+  const { status } = flowState;
+
+  const defaultSlots = React.useMemo(() => {
+    const slots = {
+      footer: showBanner ? <Footer /> : null,
```

Comment:
I'm thinking if we could simply drop `showBanner` now that we have slots API

Reply:
Agreed, it would be a breaking change though. We could deprecate it and work around it like this:
- mark as deprecated, but if used it just sends an empty slot there under the hood
- in the next major version we drop it from the API entirely